### ISSUE TITLE
[Ruby] Improve regex and heredoc recognition

### ISSUE THREAD

**Issue by zufuliu** [ScintillaOrg/lexilla#136]
Following regex or heredoc is not recognized:
1. after keyword argument or symbol Hash key `key:` (see https://docs.ruby-lang.org/en/master/syntax/methods_rdoc.html#label-Keyword+Arguments and https://docs.ruby-lang.org/en/master/syntax/literals_rdoc.html#label-Hash+Literals)
2. after method definition name.

```ruby
a = {r: /\w+/, h: <<EOF
heredoc
EOF
}

puts a

def b # :nodoc:
<<EOF
heredoc
EOF
end

def c # :nodoc:
/\w+/
end

puts b
puts c
```

[ruby-0216-preferRE.zip](https://github.com/ScintillaOrg/lexilla/files/10754961/ruby-0216-preferRE.zip)

**Comment by zufuliu** [comment]
`puts` (maybe others) can be treated same as `print`:
https://github.com/ScintillaOrg/lexilla/blob/e0ca65b31e0230fd572936b4aeb270321bd83dbe/lexers/LexRuby.cxx#L1210-L1214

**Comment by zufuliu** [comment]
[ruby-0217.zip](https://github.com/ScintillaOrg/lexilla/files/10766019/ruby-0217.zip)

Updated patch to treat `puts`, `warn` and `eval` as `print` (found with ``rg "\w+\s+<<[\w\-~'\"`]"`` on `lib\ruby` folder).

**Comment by zufuliu** [comment]
[ruby-0218.zip](https://github.com/ScintillaOrg/lexilla/files/10772251/ruby-0218.zip)

Change the code to following and leave `default`  unchanged.
```c++
case SCE_RB_WORD_DEMOTED:
case SCE_RB_DEFNAME:
case SCE_RB_IDENTIFIER_PREFERRE:
    preferRE = true;
    break;
```

**Comment by nyamatongwe** [comment]
Included in 5.2.3.
